xcm/xcm-executor/src/config.rs

```diff
@@ -106,4 +106,6 @@ pub trait Config {
 	/// Use this type to explicitly whitelist calls that cannot undergo recursion. This is a
 	/// temporary measure until we properly account for proof size weights for XCM instructions.
 	type SafeCallFilter: Contains<Self::RuntimeCall>;
+
+	type TransactionalProcessor: ProcessTransaction;
```

Comment:
why this is a configurable thing? any case we want to disable transactional processing?

Reply:
I discussed with @KiChjang that XCM should stay runtime-agnostic so it cannot have a tight coupling with Frame. That is why we made the transactional processing configurable, as other teams should be able to implement their own transactional processing.

Reply:
that does not make sense. xcm pallet are implemented with frame. 
we either have xcm spec require transactional, or not, or unspecified/implementation defined. and the implementation here follow the spec. 
so what’s the requirement from xcm spec?
presumably it is currently unspecified because xcm is poorly spec’ed and make this configurable have nothing to do with xcm runtime agnostic. there is no way to use this pallet on non frame implementation such as evm

Reply:
I think we've veered a bit off topic here. I don't think it is an expectation that every runtime MUST have transactional processing; "being able to implement their own transactional processing" also meant that they can implement no transaction processing. If it helps, we can implement `ProcessTransaction` for `()` which effectively disables it, but in most circumstances, it would most likely be advised to stick with `FrameTransactionalProcessor`.

Reply:
I am not sure if you 100% get my point.

Does XCM spec require transactional processing?
If Yes, this should not be a configurable thing.
If No, please make sure it is documented clearly in XCM spec and explain the rational.

Reply:
I think we have to correct you on one crucial factor: this is _not_ the XCM pallet. This is the XCM executor config.

Hence, it is conceivable that a non-FRAME runtime that uses substrate would still choose to integrate the XCM executor for itself. By definition, we can't make this non-configurable, as we don't know how the implementing runtime handles transactional processing, which is why I don't believe this is a spec question.

Reply:
Ok. This makes bit more sense. You should just tell me you want this code to work outside of runtime. e.g. in a lite simulator without runtime host functions.